### PR TITLE
PROD-30862: delete orphaned activity

### PR DESCRIPTION
**Pull request by denis-getopensocial**
## Problem

Some activities have no related entity after a deletion of the entity. It displays empty blocks of content.

## Solution

Create a drush command that will clean up all existing activities entity that their related entity has been deleted.

## Issue tracker

https://getopensocial.atlassian.net/browse/PROD-30862
https://getopensocial.atlassian.net/browse/PROD-30481

## How to test

- Get the database of Medtech partners on your local
- Get this branch (feature/PROD-30862-delete-orphaned-activity)
- turn on dblog: `drush en dblog`
- Run `drush cleanup-activities`
- check the logs `drush ws`
- you should see 
- `X orphaned activities with target_type "votingapi_vote" have been removed`
- `X orphaned activities with target_type "post" have been removed`
- ... 

## Change Record

Provide a command to clean up orphaned activities.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

File: modules/custom/activity_creator/src/Commands/CleanUpActivitiesDrushCommands.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    ['event_enrollment', 'id'],
+    ['groups', 'id'],
+    ['mentions', 'mid'],
+    ['node', 'nid'],
+    ['post', 'id'],
+    ['message', 'mid'],
+    ['profile','profile_id'],
+    ['queue_storage_entity', 'id'],
+    ['votingapi_vote', 'id'],
+  ];
```
Comment (denis-getopensocial) [inline modules/custom/activity_creator/src/Commands/CleanUpActivitiesDrushCommands.php:43]: is this list covering all activity types?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

File: modules/custom/activity_creator/src/Commands/CleanUpActivitiesDrushCommands.php (hunk 1 above)
Comment (robertragas) [reply modules/custom/activity_creator/src/Commands/CleanUpActivitiesDrushCommands.php]: Looks good, and if we see some activities are not cleaned up we can always add more to it.

**Comment by robertragas** [comment]
The command looks good! Only small question I have if we also want to add some output that everyone can see when running the command. 
For example the sanitize command has this.
`$this->output()->writeln(sprintf('Sanitized data for %d users.', $user_ids_count));`

As right now you don't know if it worked or not (unless you go look in the logs)
Screenshot 2024-10-01 at 14 27 05

**Review by robertragas — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4094-hudncc2jpp1agveosvwo2gt3fxntd5of.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66f57804e183914e856f4311

**Pull request merged**